### PR TITLE
fix(ec2): VPN preSharedKey cannot be a Token

### LINKED ISSUE #10723: [@aws-cdk/aws-ec2] Syntax checking of PreSharedKey for VPN
The syntax checking of PreSharedKey for VPN doesn't allow the use of like SecreValues which are represented in Token. The check really should be done only through Cloudformation.

https://github.com/aws/aws-cdk/blob/1c2cba4d34cb89e10557ccbd063c82e08b7ec025/packages/@aws-cdk/aws-ec2/lib/vpn.ts#L257

Potential fix: https://github.com/aws/aws-cdk/pull/10725

### Reproduction Steps
Creating a VPC and VPN Connection is rather a drag. Here is the important snipped:

```
const preSharedKey = SecretValue.ssmSecure('vpn-pre-shared-key', '1');
        console.log('preSharedKey = ${preSharedKey}')

        const vpnConnection = new VpnConnection(this, 'unimed-sophos', {
            ip: props.vpnConnection.customerGatewayIp,
            vpc,
            staticRoutes: props.vpnConnection.staticRoutes,
            tunnelOptions: props.vpnConnection.tunnelOptions ? [{
                preSharedKey,
            }, {
                preSharedKey,
            }] : undefined,
        });
```

### What did you expect to happen?
Secret Value is valid and throws no Error

### What actually happened?
Error:
```
The `preSharedKey` ${Token[TOKEN.158]} for tunnel 1 is invalid. Allowed characters are alphanumeric characters and ._. Must be between 8 and 64 characters in length and cannot start with zero (0).
```

### Environment

  - **CLI Version      : 1.63
  - **Framework Version:**
  - **OS               :**
  - **Language (Version):** 

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by mmuller88**
Fixes #10723

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by mmuller88** [comment]
Had some looks / thoughts regarding a test but that seems rather tricky as a proper one should include a Secret Value construct I believe. Happy if someone could help out :)

File: packages/@aws-cdk/aws-ec2/lib/vpn.ts [hunk 1]
```diff
@@ -252,12 +252,6 @@ export class VpnConnection extends cdk.Resource implements IVpnConnection {
       }
 
       props.tunnelOptions.forEach((options, index) => {
-        if (options.preSharedKey && !/^[a-zA-Z1-9._][a-zA-Z\d._]{7,63}$/.test(options.preSharedKey)) {
```
Comment (rix0rrr) [inline packages/@aws-cdk/aws-ec2/lib/vpn.ts:255]: What this needs is a `if (!Token.isUnresolved(optios.preSharedKey))` check in there, and a test.

**Review by rix0rrr — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-ec2/lib/vpn.ts (hunk 1 above)
Comment (mmuller88) [reply packages/@aws-cdk/aws-ec2/lib/vpn.ts:255]: Nice idea. Will try that

**Review by rix0rrr — APPROVED** [review]

**Comment by rix0rrr** [comment]
```
@aws-cdk/aws-ec2: [96mtest/integ.vpn-pre-shared-key-token.ts[0m:[93m14[0m:[93m11[0m - [91merror[0m[90m TS2322: [0mType 'SecretValue' is not assignable to type 'string'.
```

**Comment by mmuller88** [comment]
> ```
> @aws-cdk/aws-ec2: �[96mtest/integ.vpn-pre-shared-key-token.ts�[0m:�[93m14�[0m:�[93m11�[0m - �[91merror�[0m�[90m TS2322: �[0mType 'SecretValue' is not assignable to type 'string'.
> ```

thx @rix0rrr I fixed it :)

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**